hazelcast/src/main/java/com/hazelcast/spi/impl/NodeEngineImpl.java

```diff
@@ -234,7 +235,7 @@ public Object toObject(Object object) {
 
     @Override
     public boolean isActive() {
-        return node.isActive();
+        return node.getState() == NodeState.ACTIVE;
```

Comment:
How about moving this check into node.isActve() method? Then all other node.isActive() checks remain as they are